Add Step Functions instrumentation tips for customized way of deploying the forwarder (#21328)
DataDog/documentation: content/en/serverless/step_functions/troubleshooting.md

- For actions, we support basic actions of Lambda and DynamoDB. For example, Lambda Invoke, DynamoDB GetItem, DynamoDB PutItem, DynamoDB UpdateItem and more.
- `Wait`, `Choice`, `Success`, `Fail`, and `Pass` are supported, while `Map` and `Parallel` are not. You are able to see parallel executing spans stacked on top of each other, but no `Parallel` spans show on the flame graph.

## Customized way to deploy Datadog Lambda Forwarder
If you are using your customized way to deploy Datadog Lambda Forwarder, here are some tips that can help you debug enabling Step Functions tracing:
- On the forwarder, please set the environment variable `DD_FETCH_STEP_FUNCTIONS_TAGS` to `true`. 

> - Datadog-Forwarder layer version greater than 31 should be able to fetch state machine tags, which is a requirement for fetching `DD_TRACE_ENABLE` tag on state machines to enable Step Functions trace generation on Datadog backend.

Reviewer: I'm not sure what this means. Are you telling the user they need to have a Datadog-Forwarder layer version greater than 31?

```suggestion
- To to enable Step Functions trace generation on the Datadog backend, the Datadog-Forwarder layer version must be greater than 31. The Datadog-Forwarder fetches state machine tags, which is a requirement for fetching `DD_TRACE_ENABLE` tag on state machines.
```

Author: Yes, and the reason is that we need to get this particular tag which feature only comes after layer version 31 I believe.

Author: @estherk15 ^ thank you!

Reviewer: @kimi-p How about this?
```suggestion
- To enable Step Functions trace generation on the Datadog backend, the Datadog-Forwarder layer version must be greater than 31. This version is able to fetch state machine tags, which is a requirement for fetching the`DD_TRACE_ENABLE` tag on state machines.
```

Author: Does this make sense to you? @estherk15 
```
- To enable Step Functions trace generation on the Datadog backend, the Datadog-Forwarder layer version must be greater than 31. This version is able to fetch state machine tags, including the required `DD_TRACE_ENABLED` tag.
```

Reviewer: Nice! Yes, that's much clearer
```suggestion
- To enable Step Functions trace generation on the Datadog backend, the Datadog-Forwarder layer version must be greater than 31. This version is able to fetch state machine tags, including the required `DD_TRACE_ENABLED` tag.
```

Author: Awesome, thank you!

Author: Please help me merge once it's build.

- The IAM role for the forwarder should have `tags:getResources` permission.
- Set up a subscription filter on your state machine CloudWatch log group to the Datadog forwarder.

> - To verify if logs are reaching the Datadog backend, open the Log Explorer page and search `source:stepfunction` with the `Live` search timeframe (which will show all logs going into Datadog's logs intake). If you cannot see any logs, check if there are any error logs on the Datadog Forwarder such as wrong/invalid API key. Adding the environment variable `DD_LOG_LEVEL` of `DEBUG` helps you debug the Forwarder issue. If you see Step Functions logs, verify that the logs have the `dd_trace_enable:true` tag (all tags are normalized) and you should see Step Function traces associated with the log in a few minutes.

Reviewer: ```suggestion
- To verify if logs are reaching the Datadog backend, open the Log Explorer page and search `source:stepfunction` with the `Live` search timeframe (which shows all logs going into Datadog's logs intake). If you cannot see any logs, check if there are any error logs on the Datadog Forwarder such as wrong/invalid API key. Adding the environment variable `DD_LOG_LEVEL` of `DEBUG` helps you debug the Forwarder issue. If you see Step Functions logs, verify that the logs have the `dd_trace_enable:true` tag (all tags are normalized) and you should see Step Function traces associated with the log in a few minutes.
```
